File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.swift
import Foundation

func solution(_ ingredient:[Int]) -> Int {
    var stack: [Int] = []
    var yeon = 0
    
    for i in ingredient {
        
        stack.append(i)
        
        if stack.count < 4 { continue }
        
        let n = stack.count
        if Array(stack[n-4..<n]) == [1, 2, 3, 1] {
            stack = Array(stack[0..<n-4])
            yeon += 1
        }
        
    }
    return yeon
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack = []
    yeon = 0
    for i in ingredient:
        stack.append(i)
        if stack[-4:] == [1,2,3,1]:
            yeon += 1
            for _ in range(4):
                stack.pop()
            
    return yeon